src/ORM/DataQuery_SubGroup.php

```diff
@@ -14,26 +15,51 @@
  */
 class DataQuery_SubGroup extends DataQuery implements SQLConditionGroup
 {
+    private string $clause;
 
     /**
-     *
      * @var SQLSelect
      */
     protected $whereQuery;
 
+    /**
+     * @var SQLSelect
+     */
+    protected $havingQuery;
+
+    /**
+     * @param DataQuery $base
+     * @param string $connective
+     * @param string $clause
+     */
     public function __construct(DataQuery $base, $connective)
     {
+        // using func_get_args to add a 3rd param while retaining BC
+        // @deprecated - add a 3rd param for CMS 6 - string $clause = 'WHERE'
+        $clause = 'WHERE';
+        $args = func_get_args();
+        if (count($args) > 2) {
+            $clause = $args[2];
+        }
         parent::__construct($base->dataClass);
         $this->query = $base->query;
-        $this->whereQuery = new SQLSelect();
-        $this->whereQuery->setConnective($connective);
-
-        $base->where($this);
+        $this->clause = strtoupper($clause);
+        if ($this->clause === 'WHERE') {
+            $this->whereQuery = new SQLSelect();
+            $this->whereQuery->setConnective($connective);
+            $base->where($this);
+        } elseif ($this->clause === 'HAVING') {
+            $this->havingQuery = new SQLSelect();
+            $this->havingQuery->setConnective($connective);
+            $base->having($this);
+        } else {
+            throw new InvalidArgumentException('$clause must be either WHERE or HAVING');
+        }
     }
 
     public function where($filter)
     {
-        if ($filter) {
+        if ($filter && $this->whereQuery) {
             $this->whereQuery->addWhere($filter);
         }
```

Comment:
```suggestion
        if ($filter && $this->clause === 'WHERE') {
            $this->whereQuery->addWhere($filter);
        } else {
            throw new BadMethodCalllException('some appropriate message here');
        }
```
as per https://github.com/silverstripe/silverstripe-framework/pull/10778#discussion_r1195771967

Reply:
Same for `whereAny` and `having` methods

Reply:
Updated, though throwing LogicExceptions instead